src/render/src/encoder.rs

```diff
@@ -510,20 +510,41 @@ impl<R: Resources, C: command::Buffer<R>> Encoder<R, C> {
     pub fn clear<T: format::RenderFormat>(&mut self,
                  view: &handle::RenderTargetView<R, T>, value: T::View)
     where T::View: Into<command::ClearColor> {
-        let target = self.handles.ref_rtv(view.raw()).clone();
-        self.command_buffer.clear_color(target, value.into())
+        self.clear_raw::<T>(view.raw(), value);
     }
     /// Clear a depth view with a specified value.
     pub fn clear_depth<T: format::DepthFormat>(&mut self,
                        view: &handle::DepthStencilView<R, T>, depth: Depth) {
-        let target = self.handles.ref_dsv(view.raw()).clone();
-        self.command_buffer.clear_depth_stencil(target, Some(depth), None)
+        self.clear_depth_raw::<T>(view.raw(), depth);
     }
 
     /// Clear a stencil view with a specified value.
     pub fn clear_stencil<T: format::StencilFormat>(&mut self,
                          view: &handle::DepthStencilView<R, T>, stencil: Stencil) {
-        let target = self.handles.ref_dsv(view.raw()).clone();
+        self.clear_stencil_raw::<T>(view.raw(), stencil);
+    }
+
+
+    /// Clears the supplied `RawRenderTargetView` to the supplied `ClearColor`.
+    pub fn clear_raw<T: format::RenderFormat>(&mut self,
```

Comment:
let's just accept a `ClearColor` and no bother with passing `T` in (which is not expected for a `_raw` method)